gobblin-metrics-libs/gobblin-metrics-base/src/main/avro/GaaSObservabilityEventV0.avsc

```diff
@@ -0,0 +1,162 @@
+{
+  "type": "record",
+  "name": "GaaSObservabilityEventV0",
+  "namespace" : "org.apache.gobblin.metrics",
+  "fields": [
+    {
+      "name": "timestamp",
+      "type": "long",
+      "doc": "Time at which event was created in millis"
+    }, {
+      "name" : "flowGroup",
+      "type" : "string",
+      "doc" : "Flow group for the GaaS flow",
+      "compliance" : "NONE"
+    }, {
+      "name" : "flowName",
+      "type" : "string",
+      "doc" : "Flow name for the GaaS flow",
+      "compliance" : "NONE"
+    }, {
+      "name" : "flowExecutionId",
+      "type" : "long",
+      "doc" : "Flow execution id for the GaaS flow",
+      "compliance" : "NONE"
+    }, {
+      "name": "jobSentToExecutor",
+      "type": "boolean",
+      "doc": "Whether or not this job was able to be sent to a job executor."
+    }, {
+      "name": "lastFlowModificationTime",
+      "type": "long",
+      "doc": "Timestamp in millis when the flow config was last modified"
+    }, {
+      "name" : "edgeId",
+      "type" : "string",
+      "doc" : "Flow edge id, in format <src_node>_<dest_node>_<edge_id>",
+      "compliance" : "NONE"
+    }, {
+      "name": "jobName",
+      "type": "string",
+      "doc": "The name of the Gobblin job, found in the job template. One edge can contain multiple jobs",
+      "compliance" : "NONE"
+    }, {
+      "name": "jobType",
+      "type": {
+        "type": "enum",
+        "name": "JobType",
+        "symbols": [
+          "COPY",
+          "RETENTION",
+          "GOBBLIN"
+        ],
+        "symbolDocs": {
+          "COPY": "Gobblin distcp job",
+          "RETENTION": "Gobblin retention job",
+          "GOBBLIN": "Any Gobblin job"
+        }
+      },
+      "doc": "Gobblin job type running on GaaS, determined by the compiled job template.",
+      "compliance": "NONE"
+    }, {
+      "name": "jobStatus",
+      "type": {
+        "type": "enum",
+        "name": "JobStatus",
+        "symbols": [
+          "SUCCEEDED",
+          "FAILED",
+          "CANCELLED"
+        ],
+        "doc": "Final job status for this job in the GaaS flow",
+        "compliance": "NONE"
+      }
+    }, {
+      "name": "jobStartTime",
+      "type": "long",
+      "doc": "Start time of the job in millis",
+      "compliance": "NONE"
+    }, {
+      "name": "jobEndTime",
+      "type": "long",
+      "doc": "Finish time of the job in millis",
+      "compliance": "NONE"
+    }, {
+      "name": "proxyUser",
+      "type": "string",
+      "doc": "Proxy user (if applicable) that ran the Gobblin job",
+      "compliance": "NONE"
+    }, {
+      "name": "executorLink",
+      "type": "string",
+      "doc": "Link to where the job is running, currently limited to Azkaban",
+      "compliance": "NONE"
+    }, {
+      "name": "executorId",
+      "type": "string",
```

Comment:
would this and the executorUrl still be set in cases where the job was not orchestrated / sent to an executor, or would it then need to be `null`?  if you wish always to indicate the executor that *would* be used, since it's more where would the job execute, even if it never got submitted, then just update the doc string to clarify

Reply:
I'll set the executor url as a union of null, and the executor ID to be the executor that it would have sent to, to detect edges if they are failing to be sent to an executor.

I also wonder what to default values to if a scheduled flow fails to compile due to a flowgraph failure. It would have no executors associated with it

Reply:
great Q on compilation failure!  probably both ought to be null.

compilation error events should be purely for *scheduled flows* that previously compiled--failure at submission time should be an interactive error only, no event.  nonetheless, as we need them, perhaps we're asking `jobStatus.FAILURE` to do too much:  it's already execution failure as well as failure to send to executor--now compilation failure as well?  it becomes a status *category*, a catch-all.  better might be another status, like `INVALID` or more specific `COMPILATION_FAILURE` / `UNRESOLVED` (?).

in fact, more I consider that, I'd seek to discern `EXECUTION_FAILURE` from `SUBMISSION_FAILURE` (I dislike the overloaded terminology, since the user 'submits' a flow to gaas and then gaas 'submits' the job to the executor.)  other possible terms might be `QUEUEING_FAILURE` or `LOST` or `UNAVAILABLE`

Reply:
great Q on compilation failure! probably both ought to be null.

compilation error events should be purely for scheduled flows that previously compiled--failure at submission time should be an interactive error only, no event. nonetheless, given we need them, perhaps we're asking `jobStatus.FAILURE` to do too much: it's already execution failure as well as failure to send to executor--now compilation failure as well? it becomes a status *category*, a catch-all. better might be another status, like `INVALID` or else named `COMPILATION_FAILURE` / `UNRESOLVED` (?).

in fact, more I consider that, I'd seek also to discern `EXECUTION_FAILURE` from `SUBMISSION_FAILURE` (actually I dislike the overloaded terminology, where the user 'submits' a flow to gaas, and gaas 'submits' the job to the executor.) other possible terms might be `QUEUEING_FAILURE` or `LOST` or `UNAVAILABLE`, even `SKIPPED` if we consider possible quota overrun...

I realize that would depart from borrowing a status previously defined in the gaas job state .pdl, but faced w/ the alternative imprecision, I believe that justifiable

Reply:
I'll break up `FAILURE` to `COMPILATION_FAILURE`, `SUBMISSION_FAILURE`, and `EXECUTION_FAILURE`.
Thanks for this feedback :) We can probably update GaaS to have more statuses too but that would come in a different PR and would be a fairly major API change for users.